### PR TITLE
Update raw coprocessor

### PR DESCRIPTION
**Pull request by andylokandy**
Signed-off-by: Andy Lok <andylokandy@hotmail.com>

Issue: https://github.com/tikv/tikv/issues/9747

TiKV PR: https://github.com/tikv/tikv/pull/10229 
Rust client PR: https://github.com/tikv/client-rust/pull/293

### CODE REVIEW

**Comment by andylokandy** [comment]
cc @skyzh @Jibbow

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -993,3 +993,23 @@ message GetLockWaitInfoResponse {
...
 }
+
+message RawCoprocessorRequest {
+    kvrpcpb.Context context = 1;
+
+    string copr_name = 2;
+    // Coprorcessor version constraint according to the SEMVER definition.
+    string copr_version_req = 3;
+
+    repeated KeyRange ranges = 4;
```
Comment (skyzh) [inline proto/kvrpcpb.proto:1004]: Can we move this field to the end of the message struct in order to keep compatibility with previous proto?

**Review by skyzh — APPROVED** [review]
LGTM

File: proto/kvrpcpb.proto (hunk 1 above)
Comment (andylokandy) [reply proto/kvrpcpb.proto:1004]: Your concern is reasonable, but since the RPC has breakingly changed, it will not be worse for here.

**Comment by Jibbow** [comment]
Oh wow, thanks! :) @andylokandy

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -993,3 +993,22 @@ message GetLockWaitInfoResponse {
...
+    bytes data = 5;
+}
+
+message RawCoprocessorResponse {
+    errorpb.Error region_error = 1;
+    // Error message for cases like if no coprocessor with a matching name is found
+    // or on a version mismatch between plugin_api and the coprocessor.
+    string error = 2;
```
Comment (Jibbow) [inline proto/kvrpcpb.proto:1012]: Ah nice, I noticed as well that `other_error` doesn't play well with the macros :)

**Review by Jibbow — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Review by zhongzc — APPROVED** [review]

**Pull request merged**